### ISSUE TITLE
Add candidacy threads

### ISSUE THREAD

**Issue by traumschule** [Joystream/pioneer#4396]
I noticed that during elections a lot of communication happens on Discord that may be better saved in threads.

Similar to bounties each candidacy deserve it's own thread.

Pioneer can be modified to create a thread after announcing by batching or transmitting another tx.

The candidate card should show the number of replies and on click, either
- open the thread like normal forum thread.
- open a modal displaying the candidate node followed by responses with an option to post a reply.
- expand the candidate card and expand posts below like a proposal discussion (requires to make the card expandable like for accounts)

**Comment by dmtrjsg** [comment]
Makes sense 👍  A lot of candidates already create Forum threads created to their candidacy..
